arch/x86/core/fatal.c

```diff
@@ -155,6 +155,14 @@ static void dump_regs(const z_arch_esf_t *esf)
 #ifdef CONFIG_X86_EXCEPTION_STACK_TRACE
 	unwind_stack(esf->rbp, esf->cs);
 #endif
+#ifdef CONFIG_ASSERT
+	if (esf->rip == 0xb9) {
+		/* See implementation of __resume in locore.S. This is never
+		 * a valid RIP value.
+		 */
+		LOG_ERR("SMP: Re-use of thread object across CPUs found");
```

Comment:
Personally I'd rephrase as "Attempt to run unsuspended thread object".  The core bug here isn't that the thread object got re-used necessarily, it's that something tried to switch to it when there was no context to switch to.  It could be a garbage pointer, an uninitialized thread, or (the original use case of the poisoning) a thread that is actively running on another CPU.

Reply:
good idea